end_to_end/end_to_end_suite_test.go

```diff
@@ -2302,7 +2302,7 @@ LANGUAGE plpgsql NO SQL;`)
 			output, err := gprestoreCmd.CombinedOutput()
 			Expect(err).To(HaveOccurred())
 			Expect(string(output)).To(ContainSubstring(`Error loading data into table public.t1`))
-			Expect(string(output)).To(ContainSubstring(`Error loading data into table public.t2`))
+			Expect(string(output)).To(SatisfyAny(ContainSubstring(`Error loading data into table public.t2`), ContainSubstring(`Expected to restore 1000000 rows to table public.t2, but restored 0 instead`)))
```

Comment:
Isn't this the partial restore that I tried to avoid in the patch that added this test?

Reply:
> Isn't this the partial restore that I tried to avoid in the patch that added this test?

Commit https://github.com/arenadata/gpbackup/commit/76cd98a370dfadf39da8186c92b35f10deeb2a08
cancels all goroutines (pending on a pipe their turn to get the data from the helper), if current COPY command failed.

The changes in this PR also unblocks such pending COPY commands. So, some of them might see EOF from the pipe (resulting in the completion of the COPY command with no data restored) before their goroutines are canceled by the execution context.

The impact on the target restore database is the same in both cases, but the gprestore might show different error output.

Making error output consistent here doesn't seem good enough reason to invent additional sync mechanisms. I preferred to update the test expectations.

Reply:
I believe that partial table restore is not acceptable! The table either must be restored completely without errors, or in case of any error it must not be completely restored.

Reply:
> I believe that partial table restore is not acceptable! The table either must be restored completely without errors, or in case of any error it must not be completely restored.

After double checking, if we compare the case when pending COPY is canceled by the ExecContext and the case when the COPY finished due to EOF from a pipe:
 - `restoreSingleTableData` will return an error in both cases;
 - the table content will be the same in both cases;
 - the difference will be in error message: "Error loading data into table..." or "Expected to restore N rows to table <T>, but restored 0 instead"
 
Can you please describe the use case you are worried about? Please elaborate a scenario from the user point of view, so I could check it.

Reply:
if error occurred on one segment

when 

> pending COPY is canceled by the ExecContext

then transaction is rollback-ed, and table will be empty an all segments, while on 

> case when the COPY finished due to EOF from a pipe

table will be empty only on this one segment, i.e. partial restore

Reply:
Note: partial restore was fixed.